plugins/inputs/elasticsearch_query/README.md

```diff
@@ -54,6 +54,13 @@ Currently it is known to break on 7.x or greater versions.
     ## The date/time field in the Elasticsearch index (mandatory).
     date_field = "@timestamp"
 
+    ## If the field used for the date/time field in Elasticsearch is also using
+    ## a custom date/time format it may be required to provide the format to
+    ## correctly parse the field.
+    ##
+    ## If using one of the built in elasticsearch formats this is not required.
+    # date_field_custom_format = ""
```

Comment:
How does the user know how to write a custom format? Is this something that is common knowledge for elasticsearch users or should we describe it here or point to external docs?

I understand this was a customer request, so that customer understands how it will work. We need to make sure other existing and new users can also figure this out.

Reply:
Without the format, Elasticsearch will attempt to parse the date itself against one of its [built-in formats](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html#built-in-date-formats). It can generally do this, and that is why we have not had any bug reports about this until now. If the user purposefully creates a custom date format that Elasticsearch cannot figure out, then this field would be required. The user would need to follow the [Joda date format](https://www.elastic.co/guide/en/elasticsearch/reference/6.3/search-aggregations-bucket-daterange-aggregation.html#date-format-pattern)

I tried to add a comment that this normally should not be needed to avoid adding confusion. And not something most of our users would even need to know.

Reply:
Let's add a link to those joda format docs. If someone thinks they need the setting, it will give them enough to try it out and find for themselves that they don't need it (and not file a github issue or ask in slack how to use it.)

I think the link would fit better in the optional parameters section on line 160.

Reply:
+1